src/libmain.c

```diff
@@ -462,6 +462,18 @@ gboolean main_is_realized(void)
 }
 
 
+/**
+ *  Checks whether Geany is quitting completely right now.
+ *
+ *  @return @c TRUE if the Geany is exiting right now or @c FALSE otherwise.
+ **/
+GEANY_API_SYMBOL
+gboolean geany_is_quitting(void)
```

Comment:
I don't think you want `main_status.quitting` here - IMO you want `main_status.closing_all` instead. Note that the plugin should work both for:
- the default session (list of open files) when no project is open
- project session when some project is open

I think you can just safely rely on Geany's session management and let Geany store persistent temp files into the corresponding session and reopen them from there.

You need to use `main_status.closing_all` to detect situations when a session is being closed including all its files and this can happen either when a project is closed (for project sessions) or when Geany quits (default session).

Reply:
Maybe one more idea - wouldn't it be better to introduce this feature just as a special mode of "Instant save"? Because I think that's what this feature really is - it just picks the file names automatically, prevents some accidental tab closing but it's instant save otherwise. It would also prevent possible confusion why the Enable button in preferences gets disabled.

Reply:
Rather better would be buffer saving, all modified buffers get saved and can be restored on restart.  Saving buffers means that there is no need to do encoding conversion, or trailing space removal or any of the other modifications that are made on save and just write metadata (filename if it exists and other per document settings) as well.  That way documents both with and without names would be safe.  Might need to be in core to interact properly with existing session management, although that may also be simplified.

Reply:
initial idea was to force all temp files to be open in Geany all the time, or at least open all of them on startup. Even ones created by concurrent Geany instances or manually on disk. 
Thats why we had that function "load all temp files" and why we used "is quitting" instead of "is closing all" - to force user saving or deleting temp files after clicking "close all" and avoid dialogs only when closing window.

Now after few changes (temp files are now properly numbered by integers and are properly readable in "recent files" menu) i guess we got a better solution and this idea of keeping all existing temp files open is probably not actual.
So i did as was suggested - switched to "closing all" status and started completely relying on session management mechanism to re-open those temp files which it finds appropriate.
This seems to work fine, including general session vs project sessions.

Reply:
about extending Instantsave - this was an initial idea i tried, but then i decided to do a separate function since unification becomes too complicated both in terms of code and in terms of usage - e.g. instantsave's target directory has a bit different usecase then ours, and there are other complications.

Btw i changed mutual exclusion of instantsave vs temp_files from blocking one another's checkbox to just automatically de-activating one when another is activated. This seems a good-enough UX to me as newly created files start to look differently when you switch mode anyway

Reply:
> about extending Instantsave - this was an initial idea i tried, but then i decided to do a separate function since unification becomes too complicated both in terms of code and in terms of usage - e.g. instantsave's target directory has a bit different usecase then ours, and there are other complications.

OK, I noticed this post only now, my previous post is a bit outdated then.

I agree the semantics of the target directory are a bit different so maybe not a good idea to merge these two. But I'd still try to unify the two features a bit. First, I think the "Instant save" tab naming is quite horrible (I had to look up in Geany's documentation what it really does). So I'd suggest calling it something like "Empty files" which would contain 2 radio buttons:
- Save on creation (which would serve for instant save settings) - this would contain the target directory
- Use as Scribble - it would contain the current temp files settings

The settings in each group would be grayed-out as necessary when the other feature is enabled, default filetype combo box could be below and be shared by both.

Reply:
i guess this is doable. Not sure we necessary want to allow picking default filetype for temp files but maybe it can be useful in some case